src/ode.py

```diff
@@ -191,10 +204,31 @@ def get_temperature_loss(self, x_H_I, x_H_II, x_He_I, x_He_II, x_He_III, T, t):
         term_1 = torch.multiply(n_H_I, heating_rate_H_I) \
                  + torch.multiply(n_He_I, heating_rate_He_I) \
                  + torch.multiply(n_He_II, heating_rate_He_II)
+        term2 = 0.0
+        term3 = 0.0
+        term4 = 0.0
+
+        # [TODO] units are quite weird here. enrgy units here are in ergs while the other terms have it in eV.
+        # Apart from this, as per fukugita, the units of each coefficient are listed to be diff. too.
+        # coefficient units -> erg.cm3/s, erg/s, erg.cm3/s
+        term5 = self.n_e * (\
+        (x_H_I * self.collisional_excitation_cooling_H_I(T)) + \
+        (x_He_I * self.collisional_excitation_cooling_He_I(T)) + \
+        (x_He_II * self.collisional_excitation_cooling_He_II(T)))
+
+        # [TODO] missing nb here after dividing.......?????
+        # out unit: [ev/cm^3.s]
+        term6 = self.compton_cooling_coefficient(T)
+
+        # out unit: ??? * 1 * (1/cm^3)
+        term7 = self.free_free_cooling_coefficient(T) * (x_H_II*n_H_n_B_ratio + x_He_II*n_He_n_B_ratio + (4*x_He_III*n_He_n_B_ratio)) * self.n_e
+
+        # out unit: 1/s * ev/K * K * (1/?????) --- missing
+        term8 = 7.5 * self.hubble_parameter() * (CONSTANT_BOLTZMANN_EV * T / mu)
 
         return 4
 
-    def init_number_density_vectors(self, redshift, x_H_I, x_H_II, x_He_I, x_He_II, x_He_III):
+    def init_number_density_vectors(self, x_H_I, x_H_II, x_He_I, x_He_II, x_He_III):
```

Comment:
I think we should find a different name for this function as the term vector is misleading here (I think we talked about it before). Maybe just init_number_densities(...)?